### ISSUE TITLE
Combine PR scripts and clean up usage of merge-external-pr.sh

### ISSUE THREAD

**Issue by yenda** [status-im/status-react#3074]
This bounty has 2 requirements:

1. `./merge-external-pr.sh REPO_URL BRANCH` doesn't work (it seems like it is trying to do the push on the upstream repository). Passing in only the number of the PR does, so clean up the script to only expect that single argument so it is clearer for everyone.

2. Combine the `merge-pr.sh` and `merge-external-pr.sh` scripts, based primarily on the `external` one as it is better and clearer. This requires keeping the same checks for making sure that PRs from external contributors are verified signed by core team members and that PRs from core team members themselves are correctly signed by the developer in question. 

We are having issues checking that PRs from core team members are squashed to single commits, see [here](https://github.com/status-im/status-react/issues/3075) for ref - so make sure the same check currently working for external PRs works for ones from core team members.

The desired interface is thus: `./scripts/merge-pr.sh <PR-ID>`, which should work regardless of where the PR is hosted or if it is signed or not.

**Comment by oskarth** [comment]
@yenda Do you think we should also merge `merge-pr` script here and put bounty up for it?

**Comment by oskarth** [comment]
@andytudhope @yenda Edited issue and approve it

**Comment by status-open-bounty** [comment]
Balance: 0.000000 ETH
Tokens: SNT: 300.00
Contract address: [0x5d0016ee0cec5cb5ef84ce4952277eb0cad8b098](https://etherscan.io/address/0x5d0016ee0cec5cb5ef84ce4952277eb0cad8b098)
Network: Mainnet
Paid to: sebastiandelgado
Visit [https://openbounty.status.im](https://openbounty.status.im) to learn more.

**Comment by sebastiandelgado** [comment]
Hey all, I'm a new contributor and will spend some time today working on this.

**Comment by sebastiandelgado** [comment]
I took a shot at this with PR #3104. Thanks in advance for review/comments :)

I am making the following assumption in this PR:
 - This script will only be run by core devs who have been allowed access to external forks using the `maintainer_can_modify` ('allow edits from maintainers') rule when enabled. If this isn't the case, under which circumstances will devs without write access to upstream use this script?

I think I need the following clarifications to know if this is working as intended:
 - It seems like merge-pr.sh handles just a subset of the cases that merge-external-pr.sh, so I don't need to borrow any functionality from it. Is that right?
 - I was not able to reproduce the issue of the script asking core members to squash pull requests with single commits. Was this comment referring to the behavior of the original merge-pr.sh?
 - Given that I don't have write access to upstream, I have no way of testing that this works all the way through. Is there a way of working around this that I haven't thought of?

Thanks again!

**Comment by yenda** [comment]
Hi, thanks for taking the time to work on this issue !

> It seems like merge-pr.sh handles just a subset of the cases that merge-external-pr.sh, so I don't need to borrow any functionality from it. Is that right?

Yes, `merge-pr.sh` is used to merge core contributors PR and is mostly a simpler version of `merge-external-pr.sh`.

> I was not able to reproduce the issue of the script asking core members to squash pull requests with single commits. Was this comment referring to the behavior of the original merge-pr.sh?

What do you mean by original ? I had this issue with the latest version of merge-pr.sh

> Given that I don't have write access to upstream, I have no way of testing that this works all the way through. Is there a way of working around this that I haven't thought of ?

You can create a dummy repo to try it out on your account. We will also merge this PR with it

**Comment by andytudhope** [comment]
Thanks @sebastiandelgado! One thing, you our PR says that it will work regardless of whether the PR is signed or not, but we need to make sure that each PR gets signed by a core contributor, so that needs to be changed afaik, cc @oskarth.

**Comment by sebastiandelgado** [comment]
@yenda Thanks for clearing those up. By 'original' I meant the latest version of merge-pr.sh, so I think we're on the same page there. I have run it on a dummy repo: `https://github.com/sebastiandelgado/status-react-dummy` by modifying the script to use:
```
  : ${OWNER:=sebastiandelgado}
  : ${REPO:=status-react-dummy}
```
It worked as expected!

I did find a typo on line 83 where `$PR_LOCAL_BRANCH` is spelt as `$PR_LOCAL_BRACNCH` so I updated that but it did not change the expected outcome of the script in my tests (only PRs with a single commit were mergeable)

@andytudhope The requirement as I understood it was that the PR should work regardless of whether or not the original PR was signed by whoever wrote the code in the first place. The script does sign the the commit with the core contributors signature every time. I could have phrased that better!

I'm pushing the typo fix to the branch. If it looks good to you and I'll squash it so its ready to merge.
